arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/AccountSection.java

```diff
@@ -94,25 +96,15 @@ public AccountSection(Hub hub) {
     this.addFragment(doingAccountFragment);
   }
 
-  public void resolvePostRollback(Hub hub, MessageFrame messageFrame, CallFrame callFrame) {
+  public void resolveUponRollback(Hub hub, MessageFrame messageFrame, CallFrame callFrame) {
 
+    final AccountSnapshot preRollBackAccountSnapshot =
+        accountSnapshotAfter.deepCopy().setDeploymentInfo(hub);
     final AccountSnapshot postRollBackAccountSnapshot =
-        AccountSnapshot.canonical(hub, targetAddress);
-
```

Comment:
Taking canonical snapshots after the rollback is wrong. That snapshot has _everything_ reverted. We only want to revert the specific action performed in this section.